client/packages/system/src/Encounter/DetailView/DetailView.tsx

```diff
@@ -114,11 +114,13 @@ export const DetailView: FC = () => {
 
   const updateEncounter = useDebounceCallback(
     (patch: Partial<EncounterFragment>) =>
-      setData({
-        ...(typeof data === 'object' ? data : {}),
-        ...patch,
+      setData((prev) => {
+        return {
+          ...(typeof prev === 'object' ? prev : {}),
+          ...patch,
+        };
       }),
-    [data, setData]
+    [setData]
```

Comment:
I'm not sure you even need the `setData` in the deps array as the `setData` function should never change. Can you check if it still works correctly without it?

Reply:
nice yeah good point, thanks Carl. Will check and remove if doesn't affect functionality